firebase-dataconnect/src/main/kotlin/com/google/firebase/dataconnect/DataConnectOperationFailureResponse.kt

```diff
@@ -0,0 +1,155 @@
+  public val errors: List<ErrorInfo>
+
+  /**
+   * The successfully-decoded [rawData], if any.
+   *
+   * Will be `null` if [rawData] is `null`, or if decoding the [rawData] failed.
+   */
+  public val data: Data?
+
+  /**
+   * Returns a string representation of this object, useful for debugging.
+   *
+   * The string representation is _not_ guaranteed to be stable and may change without notice at any
+   * time. Therefore, the only recommended usage of the returned string is debugging and/or logging.
+   * Namely, parsing the returned string or storing the returned string in non-volatile storage
+   * should generally be avoided in order to be robust in case that the string representation
+   * changes.
+   *
+   * @return a string representation of this object, which includes the class name and the values of
+   * all public properties.
+   */
+  override fun toString(): String
+
+  /**
+   * Information about the error, as provided in the response payload from the backend.
+   *
+   * See https://spec.graphql.org/draft/#sec-Errors for details.
+   */
+  public interface ErrorInfo {
+    /** The error's message. */
+    public val message: String
+
+    /** The path of the field in the response data to which this error relates. */
+    public val path: List<PathSegment>
+
+    /**
+     * Compares this object with another object for equality.
+     *
+     * @param other The object to compare to this for equality.
+     * @return true if, and only if, the other object is an instance of the same implementation of
+     * [ErrorInfo] whose public properties compare equal using the `==` operator to the
+     * corresponding properties of this object.
+     */
+    override fun equals(other: Any?): Boolean
+
+    /**
+     * Calculates and returns the hash code for this object.
+     *
+     * The hash code is _not_ guaranteed to be stable across application restarts.
+     *
+     * @return the hash code for this object, that incorporates the values of this object's public
+     * properties.
+     */
+    override fun hashCode(): Int
+
+    /**
+     * Returns a string representation of this object, useful for debugging.
+     *
+     * The string representation is _not_ guaranteed to be stable and may change without notice at
+     * any time. Therefore, the only recommended usage of the returned string is debugging and/or
+     * logging. Namely, parsing the returned string or storing the returned string in non-volatile
+     * storage should generally be avoided in order to be robust in case that the string
+     * representation changes.
+     *
+     * @return a string representation of this object, suitable for logging the error indicated by
+     * this object; it will include the path formatted into a human-readable string (if the path is
+     * not empty), and the message.
+     */
+    override fun toString(): String
+
+    /** The "segment" of a path to a field in the response data. */
+    public sealed interface PathSegment {
```

Comment:
Can we move this out of the Error context(i.e not make it nested)? Its sufficiently reusable outside of an Error context.